### PR TITLE
Fix EnableAmendment tx scanning process and add tests

### PR DESCRIPTION
**Pull request by pdp2121**
## High Level Overview of Change

The amendment's ETA when reached majority or enabled was not captured by the VHS at the moment. This PR attempts to fix the issue by:

- When an `EnableAmendment` transaction is found, use ledger's `close_time` instead of transaction's date field in the `ledger` response, which does not exist (This will also simplify the code as extra `tx` request does not need to be sent to capture the date).
- Remove `date` from ledger response types transaction.
- Add test for `handleWsMessageLedgerEnableAmendments` function which scans for `EnableAmendment` transaction at Flag + 1 ledger.
- Add more logs to the amendment scanning process.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Test Plan

Leave on staging for couple days to ensure all flag+1 ledgers are scanned.

### CODE REVIEW

**Review by mvadari — COMMENTED** [review]
Can the testcases be made smaller so the files aren't so massive? Perhaps most of the unused transactions can be removed.

File: test/connections/amendment.test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  beforeAll(async () => {
+    await setupTables()
+  })
+
+  afterAll(async () => {
+    await destroy()
+  })
+
+  beforeEach(async () => {
+    await query('amendments_status').delete('*')
```
Comment (JST5000) [inline test/connections/amendment.test.ts:20]: Why are you deleting the amendments_status before and after each test?

File: test/connections/amendment.test.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    expect(amendmentStatus[0].networks).toBe('main')
+    expect(amendmentStatus[0].ledger_index).toBe(84206081)
+    expect(amendmentStatus[0].tx_hash).toBe(
+      'CA4562711E4679FE9317DD767871E90A404C7A8B84FAFD35EC2CF0231F1F6DAF',
+    )
+    expect(amendmentStatus[0].date.toISOString()).toBe(
+      '2023-11-27T14:44:30.000Z',
+    )
+    expect(amendmentStatus[0].eta).toBe(null)
+  })
```
Comment (JST5000) [inline test/connections/amendment.test.ts:70]: Should there be a test for a negative case? (e.g. someone passes in a ledger object which isn't "corrected" or isn't a flag ledger?)

File: src/shared/types/index.ts [hunk 3]
```diff
@@ -43,7 +43,6 @@ interface LedgerCorrected extends LedgerXRPL {
     (Transaction | PseudoTransaction) & {
       metaData?: TransactionMetadata
       hash: string
-      date: number
```
Comment (mvadari) [inline src/shared/types/index.ts:46]: Why was this removed?

File: test/connections/amendment.test.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    await query('amendments_status').delete('*')
+  })
+
+  test('Correctly finds EnableAmendment tx with tfGotMajority Flag (eta available) from ledger response', async () => {
+    await handleWsMessageLedgerEnableAmendments(
+      ledgerResponseGotMajority as LedgerResponseCorrected,
+      'main',
+    )
+
+    await flushPromises()
```
[the remark this answers is not in the document]
Comment (mvadari) [reply test/connections/amendment.test.ts:33]: It makes sure all the awaited functions have fully run (just to prevent race conditions).

**Review by mvadari — APPROVED** [review]

File: test/connections/amendment.test.ts (hunk 1 above)
Comment (pdp2121) [reply test/connections/amendment.test.ts:20]: It's to clean up the amendments_status table in the db between each test otherwise the next test might have data from the previous one

File: src/shared/types/index.ts (hunk 3 above)
Comment (pdp2121) [reply src/shared/types/index.ts:46]: This was added by mistake as I thought there's a date field in the expanded ledger response but actually there isn't and we can use the `close_time` instead

**Review by JST5000 — APPROVED** [review]
LGTM
